include/Bezier/bezier.h

```diff
@@ -246,23 +257,46 @@ class Curve
    * \param epsilon Precision of resulting intersection
    * \return A vector af points of intersection between curves
    */
-  PointVector pointsOfIntersection(const Curve& curve, bool stop_at_first = false, double epsilon = 0.001) const;
+  PointVector intersection(const Curve& curve, bool stop_at_first = false, double epsilon = 0.001) const;
 
   /*!
    * \brief Get the parameter t where curve is closest to given point
    * \param point Point to project on curve
-   * \param step Size of step in coarse search
    * \param epsilon Precision of resulting projection
    * \return Parameter t
    */
-  double projectPoint(const Point& point, double step = 0.01, double epsilon = 0.001, std::size_t max_iter = 15) const;
+  Parameter projectPoint(const Point& point, double epsilon = 0.001) const;
+
+  /*!
+   * \brief Get the parameter t vector where curve is closest to given points
+   * \param point_vector Points to project on curve
+   * \param epsilon Precision of resulting projection
+   * \return Vector of parameters t
+   */
+  ParameterVector projectPoint(PointVector point_vector, double epsilon = 0.001) const;
+
+  /*!
+   * \brief Get distance of the point to the curve
+   * \param point Point to project on curve
+   * \param epsilon Precision of resulting projection
+   * \return Distance to the curve
+   */
+  double distance(const Point& point, double epsilon = 0.001) const;
+
+  /*!
+   * \brief Get the distance vector of points to the curve
+   * \param point_vector Points to project on curve
+   * \param epsilon Precision of resulting projection
+   * \return Vector of distances
+   */
+  std::vector<double> distance(PointVector point_vector, double epsilon = 0.001) const;
 
   /*!
    * \brief applyContinuity Apply geometric continuity based on the another curve.
    * \param locked_curve Curve on which calculation are based.
    * \param beta_coeffs Beta-constraints used to calculate continuity. Size defines continuity order.
    */
-  void applyContinuity(const Curve& source_curve, std::vector<double>& beta_coeffs);
+  void applyContinuity(const Curve& source_curve, ParameterVector& beta_coeffs);
```

Comment:
Beta coefficients are not the same as t parameter. It should remain as it was.

Reply:
this was a mistake on my part, will fix